Mlem/Views/Tabs/Settings/Components/Settings View.swift

```diff
@@ -10,83 +10,86 @@ import SwiftUI
+                        }
+                        
+                        NavigationLink(value: SettingsRoute.appearance) {
+                            Label("Appearance", systemImage: "paintbrush.fill").labelStyle(SquircleLabelStyle(color: .pink))
+                        }
+                        
+                        NavigationLink(value: SettingsRoute.contentFilters) {
+                            Label("Content Filters", systemImage: "line.3.horizontal.decrease")
+                                .labelStyle(SquircleLabelStyle(color: .orange))
+                        }
                     }
                     
-                    NavigationLink {
-                        AccessibilitySettingsView()
-                    } label: {
-                        // apparently the Apple a11y symbol isn't an SFSymbol
-                        Label("Accessibility", systemImage: "hand.point.up.braille.fill").labelStyle(SquircleLabelStyle(color: .blue))
+                    Section {
+                        NavigationLink(value: SettingsRoute.about) {
+                            Label("About Mlem", systemImage: "info").labelStyle(SquircleLabelStyle(color: .blue))
+                        }
                     }
                     
-                    NavigationLink {
-                        AppearanceSettingsView()
-                    } label: {
-                        Label("Appearance", systemImage: "paintbrush.fill").labelStyle(SquircleLabelStyle(color: .pink))
-                    }
-
-                    NavigationLink {
-                        FiltersSettingsView()
-                    } label: {
-                        Label("Content Filters", systemImage: "line.3.horizontal.decrease").labelStyle(SquircleLabelStyle(color: .orange))
-                    }
-                }
-                
-                Section {
-                    NavigationLink {
-                        AboutView(navigationPath: $navigationPath)
-                    } label: {
-                        Label("About Mlem", systemImage: "info").labelStyle(SquircleLabelStyle(color: .blue))
+                    Section {
+                        NavigationLink(value: SettingsRoute.advanced) {
+                            Label("Advanced", systemImage: "gearshape.2.fill").labelStyle(SquircleLabelStyle(color: .gray))
+                        }
                     }
                 }
-                
-                Section {
-                    NavigationLink {
-                        AdvancedSettingsView()
-                    } label: {
-                        Label("Advanced", systemImage: "gearshape.2.fill").labelStyle(SquircleLabelStyle(color: .gray))
+                .onChange(of: selectedNavigationTabHashValue) { newValue in
+                    if newValue == TabSelection.settings.hashValue {
+                        print("re-selected \(TabSelection.settings) tab")
+#if DEBUG
+                        if navigationPath.isEmpty {
```

Comment:
Personally I really like the idea of that behavior, but I think it absolutely needs to be brought in with #174--it's bad UX IMO to have an easily-triggered way to go back without an easy way to undo it.

Reply:
> what are your thoughts on the behaviour here being that the first tap scrolls to top, and if we're already at the top we pop back? (replacing the `.isEmpty` check with an _at top_ check effectively)

Yea, that's definitely something I see users wanting, but should be an option. I see three options:
Option 1. Tap to go back to root (skip all other pages).
Option 2. Tap to go back to each page (don't scroll to top unless at root page).
Option 3. "Zig-Zag" Pop->Scroll->Pop->Scroll

I swear I thought the system iOS behaviour used to be Option 2, but it doesn't seem to be the case (anymore?).

Reply:
> Personally I really like the idea of that behavior, but I think it absolutely needs to be brought in with #174--it's bad UX IMO to have an easily-triggered way to go back without an easy way to undo it.

I'd be okay with this being shipped, but **turned off** by default until we figure out #174 and "tap tab again to scroll back to previous scroll position". The tap tab behaviour seems pretty well requested, and the system behaviour doesn't allow you to swipe forward anyway =P

Reply:
> Good point, we need to figure that one out before adding it into the feed tab 👍
> 
> I think adding it here in settings is safe enough, but I guess it could be argued that until we're ready to roll it out across all the tabs it'd be sensible to keep the current one behind the `#if DEBUG` condition?
> 
> Saves people wondering why it works on one and not others 🙃

I totally didn't start with Settings because it'd be lower stakes ahaha XD
Imagine all the bug reports we'd get if we shipped it just in Settings..."muh Tab Bar is borken!"